website_blog_group_old_archives/controllers/website_blog.py

```diff
@@ -0,0 +1,75 @@
+# -*- coding: utf-8 -*-
+# © 2016 Therp BV <http://therp.nl>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import datetime
+from openerp import http, tools
+from openerp.http import request
+from openerp.addons.website_blog.controllers.main import WebsiteBlog
+
+class website_blog(WebsiteBlog):
+
+
+    def nav_list_grouped(self):
+        blog_post_obj = request.registry['blog.post']
+        beginning_of_year = datetime.date.strftime(
+            datetime.date(datetime.date.today().year, 1, 1),
+            tools.DEFAULT_SERVER_DATE_FORMAT
+        )
+        old_groups = blog_post_obj.read_group(
+            request.cr, request.uid, 
+            [['create_date', '<', beginning_of_year]],
+            ['name', 'create_date'],
+            groupby="create_date:year", 
+            orderby="create_date:year desc", 
+            context=request.context
+        )       
+        # the standard orderby is just a comma separated list of fields
+        # optionally followed by desc or asc.
+        # UNDOCUMENTED FEATURE, the time :year, :month, :day work also 
+        # for orderby.
+        new_groups = blog_post_obj.read_group(
+            request.cr, request.uid,
+            [['create_date', '>=', beginning_of_year]],
+            ['name', 'create_date'],
+            groupby="create_date", 
+            orderby="create_date desc", 
+            context=request.context
+        )
+        for group in new_groups:
+            begin_date = datetime.datetime.strptime(
```

Comment:
all of those can be simplified a lot by using `openerp.fields.Date{,time}.from_string`